rpm/0002-Accounts-qt-integration.patch

```diff
@@ -84,1966 +84,74 @@ This mirrors the behavour of qmf-oauth2-plugin. See
-+        }
-+    }
- 
-+    result->setId(id);
-+    result->setModified(false);
-+
-+    return Success;
-+}
-+#else
- QMailStorePrivate::AttemptResult QMailStorePrivate::attemptAccountConfiguration(const QMailAccountId &id, 
-                                                                                 QMailAccountConfiguration *result, 
-                                                                                 ReadLock &)
-@@ -7742,6 +8824,7 @@ QMailStorePrivate::AttemptResult QMailStorePrivate::attemptAccountConfiguration(
- 
-     return Success;
- }
-+#endif
- 
- QMailStorePrivate::AttemptResult QMailStorePrivate::attemptThread(const QMailThreadId &id, QMailThread *result, ReadLock &)
- {
-@@ -8860,11 +9943,26 @@ bool QMailStorePrivate::checkPreconditions(const QMailFolder& folder, bool updat
- 
-     if(folder.parentAccountId().isValid())
-     {
-+#ifdef USE_ACCOUNTS_QT
-+        Accounts::AccountId accountId = folder.parentAccountId().toULongLong();
-+        QSharedPointer<Accounts::Account> ssoAccount = getEmailAccount(accountId);
-+
-+        if (!ssoAccount) {
-+            qWarning() << "Parent account does not exist!";
-+            return false;
-+        }
-+
-+        if (!ssoAccount->supportsService(QLatin1String("e-mail"))) {
-+            qMailLog(Messaging) << "Parent account does not support e-mail service!";
-+            return false;
-+        }
-+#else
-         if (!idExists(folder.parentAccountId(), QLatin1String("mailaccounts")))
+diff --git a/src/libraries/qmfclient/qmailaccountlistmodel.cpp b/src/libraries/qmfclient/qmailaccountlistmodel.cpp
+index 28f566cb..63d98c53 100644
+--- a/src/libraries/qmfclient/qmailaccountlistmodel.cpp
++++ b/src/libraries/qmfclient/qmailaccountlistmodel.cpp
+@@ -341,31 +341,37 @@ void QMailAccountListModel::accountsAdded(const QMailAccountIdList& ids)
+     { 
+         foreach(const QMailAccountId &id,results)
          {
-             qWarning() << "Parent account does not exist!";
-             return false;
-         }
-+#endif
-     }
- 
-     return true;
-@@ -9387,6 +10485,14 @@ bool QMailStorePrivate::deleteAccounts(const QMailAccountKey& key,
-                                        QMailThreadIdList& modifiedThreadIds,
-                                        QMailAccountIdList& modifiedAccountIds)
- {
-+#ifdef USE_ACCOUNTS_QT
-+    // Searching across all accounts inside SSO
-+    deletedAccountIds << searchSSOAccounts(key);
-+
-+    // No accounts? Then we're already done
-+    if (deletedAccountIds.isEmpty())
-+        return true;
-+#else
-     {
-         // Get the identifiers for all the accounts we're deleting
-         QSqlQuery query(simpleQuery(QLatin1String("SELECT t0.id FROM mailaccounts t0"),
-@@ -9406,6 +10512,7 @@ bool QMailStorePrivate::deleteAccounts(const QMailAccountKey& key,
-         if (noAccounts)
-             return true;
-     }
-+#endif
+-            LessThanFunctorA lessThan(d->sortKey);
++            if (!d->idList.contains(id)) {
```

Comment:
No need to adjust now, but just noting that even the remaining delta could be smaller by making the contains() checks/filtering first and then just keeping the upstream handler. Or wondering if even upstream should be having some of that. Enough changes even without that but just saying :)

Reply:
Indeed, I'll treat the remaining chunks in this patch in a seoparate PR, just focussed on what is left. I've adjusted the description of the patch accordingly.